tests/ui/fn/param-mismatch-trait-fn.stderr

```diff
@@ -0,0 +1,23 @@
+error[E0308]: mismatched types
+  --> $DIR/param-mismatch-trait-fn.rs:6:23
+   |
+LL |     T::same_type([x], Some(y));
+   |     ------------ ---  ^^^^^^^ expected `[X; 1]`, found `Option<Y>`
+   |     |            |
+   |     |            expected all arguments to be this `[X; 1]` type because they need to match the type of this parameter
+   |     arguments to this function are incorrect
+   |
+   = note: expected array `[X; 1]`
+               found enum `Option<Y>`
+note: associated function defined here
+  --> $DIR/param-mismatch-trait-fn.rs:2:8
+   |
+LL |     fn same_type<T>(_: T, _: T);
+   |        ^^^^^^^^^ -  -     - this parameter needs to match the `[X; 1]` type of ``
+   |                  |  |
+   |                  |  `` needs to match the `[X; 1]` type of this parameter
+   |                  `` and `` both reference this parameter `T`
```

Comment:
btw, are all these empty `` names expected?

Reply:
Whoops, yeah we should definitely be returning `None` if the param names are empty.